### ISSUE TITLE
No Data from myVaillant after migration from Multimatic App.

### ISSUE THREAD

**Issue by Brudertac** [signalkraft/mypyllant-component#92]
### Before submitting a new issue

- [ x ] I'm not using the older Vaillant sensoAPP (which is unsupported by this integration)
- [ x ] I [redownloaded the latest release](https://my.home-assistant.io/redirect/hacs_repository/?owner=signalkraft&repository=mypyllant-component) and restarted Home Assistant
- [ x ] I've [enabled debug logs](https://signalkraft.com/mypyllant-component/3-contributing/#debugging) and attached the relevant logs to this ticket

### Problem description

Hello and before i start. Thanks for this!

Just migrated from the Multimatic App to myVaillant. The myVaillant App shows all Settings and so i activated mypyllant...
But it is not loading Data from the myVaillant API.

### Logs

```
2024-02-06 15:36:49.330 WARNING (SyncWorker_3) [homeassistant.loader] We found a custom integration hacs which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant
2024-02-06 15:36:49.333 WARNING (SyncWorker_3) [homeassistant.loader] We found a custom integration multimatic which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant
2024-02-06 15:36:49.335 WARNING (SyncWorker_3) [homeassistant.loader] We found a custom integration mypyllant which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant
2024-02-06 15:37:04.870 WARNING (MainThread) [myPyllant.api] Could not get RTS data
Traceback (most recent call last):
  File "/usr/local/lib/python3.11/site-packages/myPyllant/api.py", line 1022, in get_rts
    response = await self.aiohttp_session.get(
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/aiohttp/client.py", line 691, in _request
    await raise_for_status(resp)
  File "/usr/local/lib/python3.11/site-packages/myPyllant/api.py", line 102, in on_raise_for_status
    response.raise_for_status()
  File "/usr/local/lib/python3.11/site-packages/aiohttp/client_reqrep.py", line 1060, in raise_for_status
    raise ClientResponseError(
aiohttp.client_exceptions.ClientResponseError: 404, message='Resource Not Found', url=URL('https://api.vaillant-group.com/service-connected-control/vrc700/v1/rts/...CUT.../devices')
2024-02-06 15:37:04.912 ERROR (MainThread) [custom_components.mypyllant] Unexpected error fetching myVAILLANT data: 'DAY' is not a valid DHWOperationModeVRC700
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/update_coordinator.py", line 300, in _async_refresh
    self.data = await self._async_update_data()
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/mypyllant/__init__.py", line 350, in _async_update_data
    data = [
           ^
  File "/config/custom_components/mypyllant/__init__.py", line 350, in <listcomp>
    data = [
           ^
  File "/usr/local/lib/python3.11/site-packages/myPyllant/api.py", line 386, in get_systems
    system = System.from_api(
             ^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/myPyllant/models.py", line 554, in from_api
    system.domestic_hot_water = [
                                ^
  File "/usr/local/lib/python3.11/site-packages/myPyllant/models.py", line 555, in <listcomp>
    DomesticHotWater.from_api(
  File "/usr/local/lib/python3.11/site-packages/myPyllant/models.py", line 490, in from_api
    kwargs["operation_mode_dhw"] = DHWOperationModeVRC700(
                                   ^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/enum.py", line 712, in __call__
    return cls.__new__(cls, value)
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/enum.py", line 1135, in __new__
    raise ve_exc
ValueError: 'DAY' is not a valid DHWOperationModeVRC700
2024-02-06 15:37:05.292 ERROR (MainThread) [custom_components.mypyllant] Unexpected error fetching myVAILLANT data: 'DAY' is not a valid DHWOperationModeVRC700
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/update_coordinator.py", line 300, in _async_refresh
    self.data = await self._async_update_data()
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/mypyllant/__init__.py", line 379, in _async_update_data
    async for system in await self.hass.async_add_executor_job(
  File "/usr/local/lib/python3.11/site-packages/myPyllant/api.py", line 386, in get_systems
    system = System.from_api(
             ^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/myPyllant/models.py", line 554, in from_api
    system.domestic_hot_water = [
                                ^
  File "/usr/local/lib/python3.11/site-packages/myPyllant/models.py", line 555, in <listcomp>
    DomesticHotWater.from_api(
  File "/usr/local/lib/python3.11/site-packages/myPyllant/models.py", line 490, in from_api
    kwargs["operation_mode_dhw"] = DHWOperationModeVRC700(
                                   ^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/enum.py", line 712, in __call__
    return cls.__new__(cls, value)
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/enum.py", line 1135, in __new__
    raise ve_exc
ValueError: 'DAY' is not a valid DHWOperationModeVRC700
2024-02-06 15:37:05.315 WARNING (MainThread) [custom_components.mypyllant.binary_sensor] No system data, skipping binary sensors
2024-02-06 15:37:05.316 WARNING (MainThread) [custom_components.mypyllant.sensor] No system data, skipping sensors
2024-02-06 15:37:05.317 WARNING (MainThread) [custom_components.mypyllant.sensor] No daily data, skipping sensors
2024-02-06 15:37:05.401 WARNING (MainThread) [custom_components.mypyllant.climate] No system data, skipping climate
2024-02-06 15:37:05.404 WARNING (MainThread) [custom_components.mypyllant.number] No system data, skipping number entities
2024-02-06 15:37:05.406 WARNING (MainThread) [custom_components.mypyllant.datetime] No system data, skipping date time entities
2024-02-06 15:37:05.407 WARNING (MainThread) [custom_components.mypyllant.calendar] No system data, skipping calendar entities
2024-02-06 15:37:05.409 WARNING (MainThread) [custom_components.mypyllant.water_heater] No system data, skipping water heater
2024-02-06 15:37:05.410 WARNING (MainThread) [custom_components.mypyllant.switch] No system data, skipping switch entities
2024-02-06 15:42:41.363 ERROR (MainThread) [frontend.js.latest.202401040] Uncaught error from Edge 121.0.0.0 on Windows 10
TypeError: Cannot read properties of undefined (reading 'takeControl')
_takeControl (src/panels/lovelace/strategies/device-registry-detail/dialog-dashboard-strategy-editor.ts:176:18)
call (src/lit-html.ts:2018:28)
dispatchEvent (mwc-list-item-base.ts:247:9)
fireRequestSelected (mwc-list-item-base.ts:92:9)
call (observer.ts:49:23)
forEach (observer.ts:42:28)
updated (src/reactive-element.ts:1378:9)
_$AE (src/reactive-element.ts:1343:11)
performUpdate (src/reactive-element.ts:1261:16)
2024-02-06 15:43:24.279 DEBUG (MainThread) [custom_components.mypyllant] Starting mypyllant component v0.7.2 (library 0.7.7) with homeassistant 2024.1.6, dacite 1.8.1, and aiohttp 3.9.3
2024-02-06 15:43:24.280 DEBUG (MainThread) [custom_components.mypyllant] Creating API and logging in with ...CUT... in realm germany
2024-02-06 15:43:27.505 DEBUG (MainThread) [custom_components.mypyllant] Refreshing SystemCoordinator
2024-02-06 15:43:27.505 DEBUG (MainThread) [custom_components.mypyllant] Starting async update data for SystemCoordinator
2024-02-06 15:43:27.505 DEBUG (MainThread) [custom_components.mypyllant] Waiting 119s until token refresh for ...CUT...
2024-02-06 15:43:28.999 WARNING (MainThread) [myPyllant.api] Could not get RTS data
Traceback (most recent call last):
  File "/usr/local/lib/python3.11/site-packages/myPyllant/api.py", line 1022, in get_rts
    response = await self.aiohttp_session.get(
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/aiohttp/client.py", line 691, in _request
    await raise_for_status(resp)
  File "/usr/local/lib/python3.11/site-packages/myPyllant/api.py", line 102, in on_raise_for_status
    response.raise_for_status()
  File "/usr/local/lib/python3.11/site-packages/aiohttp/client_reqrep.py", line 1060, in raise_for_status
    raise ClientResponseError(
aiohttp.client_exceptions.ClientResponseError: 404, message='Resource Not Found', url=URL('https://api.vaillant-group.com/service-connected-control/vrc700/v1/rts/...CUT.../devices')
2024-02-06 15:43:29.020 ERROR (MainThread) [custom_components.mypyllant] Unexpected error fetching myVAILLANT data: 'DAY' is not a valid DHWOperationModeVRC700
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/update_coordinator.py", line 300, in _async_refresh
    self.data = await self._async_update_data()
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/mypyllant/__init__.py", line 350, in _async_update_data
    data = [
           ^
  File "/config/custom_components/mypyllant/__init__.py", line 350, in <listcomp>
    data = [
           ^
  File "/usr/local/lib/python3.11/site-packages/myPyllant/api.py", line 386, in get_systems
    system = System.from_api(
             ^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/myPyllant/models.py", line 554, in from_api
    system.domestic_hot_water = [
                                ^
  File "/usr/local/lib/python3.11/site-packages/myPyllant/models.py", line 555, in <listcomp>
    DomesticHotWater.from_api(
  File "/usr/local/lib/python3.11/site-packages/myPyllant/models.py", line 490, in from_api
    kwargs["operation_mode_dhw"] = DHWOperationModeVRC700(
                                   ^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/enum.py", line 712, in __call__
    return cls.__new__(cls, value)
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/enum.py", line 1135, in __new__
    raise ve_exc
ValueError: 'DAY' is not a valid DHWOperationModeVRC700
2024-02-06 15:43:29.026 DEBUG (MainThread) [custom_components.mypyllant] Finished fetching myVAILLANT data in 1.521 seconds (success: False)
2024-02-06 15:43:29.027 DEBUG (MainThread) [custom_components.mypyllant] Refreshing DailyDataCoordinator
2024-02-06 15:43:29.027 DEBUG (MainThread) [custom_components.mypyllant] Starting async update data for DailyDataCoordinator
2024-02-06 15:43:29.027 DEBUG (MainThread) [custom_components.mypyllant] Waiting 118s until token refresh for ...CUT...
2024-02-06 15:43:29.355 ERROR (MainThread) [custom_components.mypyllant] Unexpected error fetching myVAILLANT data: 'DAY' is not a valid DHWOperationModeVRC700
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/update_coordinator.py", line 300, in _async_refresh
    self.data = await self._async_update_data()
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/mypyllant/__init__.py", line 379, in _async_update_data
    async for system in await self.hass.async_add_executor_job(
  File "/usr/local/lib/python3.11/site-packages/myPyllant/api.py", line 386, in get_systems
    system = System.from_api(
             ^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/myPyllant/models.py", line 554, in from_api
    system.domestic_hot_water = [
                                ^
  File "/usr/local/lib/python3.11/site-packages/myPyllant/models.py", line 555, in <listcomp>
    DomesticHotWater.from_api(
  File "/usr/local/lib/python3.11/site-packages/myPyllant/models.py", line 490, in from_api
    kwargs["operation_mode_dhw"] = DHWOperationModeVRC700(
                                   ^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/enum.py", line 712, in __call__
    return cls.__new__(cls, value)
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/enum.py", line 1135, in __new__
    raise ve_exc
ValueError: 'DAY' is not a valid DHWOperationModeVRC700
2024-02-06 15:43:29.363 DEBUG (MainThread) [custom_components.mypyllant] Finished fetching myVAILLANT data in 0.336 seconds (success: False)
2024-02-06 15:43:29.367 WARNING (MainThread) [custom_components.mypyllant.binary_sensor] No system data, skipping binary sensors
2024-02-06 15:43:29.367 WARNING (MainThread) [custom_components.mypyllant.calendar] No system data, skipping calendar entities
2024-02-06 15:43:29.367 WARNING (MainThread) [custom_components.mypyllant.climate] No system data, skipping climate
2024-02-06 15:43:29.367 WARNING (MainThread) [custom_components.mypyllant.datetime] No system data, skipping date time entities
2024-02-06 15:43:29.367 WARNING (MainThread) [custom_components.mypyllant.number] No system data, skipping number entities
2024-02-06 15:43:29.367 WARNING (MainThread) [custom_components.mypyllant.sensor] No system data, skipping sensors
2024-02-06 15:43:29.367 DEBUG (MainThread) [custom_components.mypyllant.sensor] Daily data: None
2024-02-06 15:43:29.368 WARNING (MainThread) [custom_components.mypyllant.sensor] No daily data, skipping sensors
2024-02-06 15:43:29.368 WARNING (MainThread) [custom_components.mypyllant.switch] No system data, skipping switch entities
2024-02-06 15:43:29.368 WARNING (MainThread) [custom_components.mypyllant.water_heater] No system data, skipping water heater
```

**Comment by Strohhutpat** [comment]
Same

**Comment by signalkraft** [comment]
Please check out the latest pre-release, I added the missing value. If you get more errors, please post them. The Vaillant API isn't documented, and I don't have the same setup as you verify changes.

**Comment by Brudertac** [comment]
Thank you. With 0.7.3b0 it works fine.

**Comment by MrBuff** [comment]
First a big thank you for this integration.

Yesterday i migrated from the MultiMatic to the myVaillant app. The myVaillant app works like it should, but when i try to install mypyllant, i can't see anything. I tired 0.72 and also 0.73b but in both cases i get the following errors in the log:

2024-02-07 12:30:10.263 ERROR (MainThread) [custom_components.mypyllant] Unexpected error fetching myVAILLANT data: 'types.UnionType' object has no attribute '__origin__'
Traceback (most recent call last):
  File "/usr/local/lib/python3.11/site-packages/dacite/types.py", line 22, in extract_origin_collection
    return collection.__extra__
           ^^^^^^^^^^^^^^^^^^^^
AttributeError: 'types.UnionType' object has no attribute '__extra__'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/update_coordinator.py", line 300, in _async_refresh
    self.data = await self._async_update_data()
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/mypyllant/__init__.py", line 350, in _async_update_data
    data = [
           ^
  File "/config/custom_components/mypyllant/__init__.py", line 350, in <listcomp>
    data = [
           ^
  File "/usr/local/lib/python3.11/site-packages/myPyllant/api.py", line 365, in get_systems
    async for home in homes:
  File "/usr/local/lib/python3.11/site-packages/myPyllant/api.py", line 339, in get_homes
    yield Home.from_api(timezone=timezone, **home_json)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/myPyllant/models.py", line 69, in from_api
    return from_dict(
           ^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/dacite/core.py", line 64, in from_dict
    value = _build_value(type_=field_type, data=field_data, config=config)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/dacite/core.py", line 101, in _build_value
    if is_subclass(type_, cast_type):
       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/dacite/types.py", line 168, in is_subclass
    if is_generic_collection(sub_type):
       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/dacite/types.py", line 147, in is_generic_collection
    origin = extract_origin_collection(type_)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/dacite/types.py", line 24, in extract_origin_collection
    return collection.__origin__
           ^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'types.UnionType' object has no attribute '__origin__'
2024-02-07 12:30:10.427 ERROR (MainThread) [custom_components.mypyllant] Unexpected error fetching myVAILLANT data: 'types.UnionType' object has no attribute '__origin__'
Traceback (most recent call last):
  File "/usr/local/lib/python3.11/site-packages/dacite/types.py", line 22, in extract_origin_collection
    return collection.__extra__
           ^^^^^^^^^^^^^^^^^^^^
AttributeError: 'types.UnionType' object has no attribute '__extra__'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/update_coordinator.py", line 300, in _async_refresh
    self.data = await self._async_update_data()
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/mypyllant/__init__.py", line 379, in _async_update_data
    async for system in await self.hass.async_add_executor_job(
  File "/usr/local/lib/python3.11/site-packages/myPyllant/api.py", line 365, in get_systems
    async for home in homes:
  File "/usr/local/lib/python3.11/site-packages/myPyllant/api.py", line 339, in get_homes
    yield Home.from_api(timezone=timezone, **home_json)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/myPyllant/models.py", line 69, in from_api
    return from_dict(
           ^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/dacite/core.py", line 64, in from_dict
    value = _build_value(type_=field_type, data=field_data, config=config)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/dacite/core.py", line 101, in _build_value
    if is_subclass(type_, cast_type):
       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/dacite/types.py", line 168, in is_subclass
    if is_generic_collection(sub_type):
       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/dacite/types.py", line 147, in is_generic_collection
    origin = extract_origin_collection(type_)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/dacite/types.py", line 24, in extract_origin_collection
    return collection.__origin__
           ^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'types.UnionType' object has no attribute '__origin__'
2024-02-07 12:30:10.699 WARNING (MainThread) [custom_components.mypyllant.binary_sensor] No system data, skipping binary sensors
2024-02-07 12:30:10.699 WARNING (MainThread) [custom_components.mypyllant.calendar] No system data, skipping calendar entities
2024-02-07 12:30:10.700 WARNING (MainThread) [custom_components.mypyllant.climate] No system data, skipping climate
2024-02-07 12:30:10.700 WARNING (MainThread) [custom_components.mypyllant.number] No system data, skipping number entities
2024-02-07 12:30:10.701 WARNING (MainThread) [custom_components.mypyllant.sensor] No system data, skipping sensors
2024-02-07 12:30:10.701 WARNING (MainThread) [custom_components.mypyllant.sensor] No daily data, skipping sensors
2024-02-07 12:30:10.701 WARNING (MainThread) [custom_components.mypyllant.switch] No system data, skipping switch entities
2024-02-07 12:30:10.702 WARNING (MainThread) [custom_components.mypyllant.water_heater] No system data, skipping water heater
2024-02-07 12:30:10.811 WARNING (MainThread) [custom_components.mypyllant.datetime] No system data, skipping date time entities`

**Comment by switschel** [comment]
Same here. using v7.3.0b0. Removed govee and ended up with this error in the log.
Setup: 
- aroTherm plus 75/8
- VRC921
- VR700

`Traceback (most recent call last):
  File "/usr/local/lib/python3.12/site-packages/dacite/core.py", line 72, in from_dict
    value = get_default_value_for_field(field, field_type)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/dacite/dataclasses.py", line 21, in get_default_value_for_field
    raise DefaultValueNotFoundError()
dacite.dataclasses.DefaultValueNotFoundError

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/update_coordinator.py", line 313, in _async_refresh
    self.data = await self._async_update_data()
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/mypyllant/__init__.py", line 350, in _async_update_data
    data = [
           ^
  File "/usr/local/lib/python3.12/site-packages/myPyllant/api.py", line 386, in get_systems
    system = System.from_api(
             ^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/myPyllant/models.py", line 542, in from_api
    Zone.from_api(
  File "/usr/local/lib/python3.12/site-packages/myPyllant/models.py", line 386, in from_api
    ZoneCooling.from_api(**kwargs["cooling"]) if "cooling" in kwargs else None
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/myPyllant/models.py", line 313, in from_api
    return super().from_api(**kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/myPyllant/models.py", line 69, in from_api
    return from_dict(
           ^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/dacite/core.py", line 76, in from_dict
    raise MissingValueError(field.name)
dacite.exceptions.MissingValueError: missing value for field "manual_mode_setpoint_cooling"`

**Comment by MichelGuiaux** [comment]
Thank you. With 0.7.3b0 it works fine for me too

**Comment by Brudertac** [comment]
@signalkraft 

how can i assist? Using 0.7.3b0 and the System will not Load new Data from Vaillant. I also have enabled Debugging and see some Errors in the Logs:

[home-assistant_2024-02-08T08-32-19.352Z.log](https://github.com/signalkraft/mypyllant-component/files/14206326/home-assistant_2024-02-08T08-32-19.352Z.log)

**Comment by signalkraft** [comment]
Could you try the latest pre-release?

**Comment by m7r** [comment]
@signalkraft after 0.7.3b1 the errors in myVAILLANT debug log are gone.

I have an issue with hotwater boost it is possible to enable and disable it. But the switch goes automatically to off even oft the boost is still active.
workaround: enable it again and immediately stop it.

**Comment by 85Kid** [comment]
I use the latest release. Since then the integration fetches the data only once after a reload.

**Comment by switschel** [comment]
Seems to work with 0.7.3b1.
I recognized that there are now engery data fetched even it exists.
In the log I still have this error:

Error fetching myVAILLANT data: 404, message='Resource Not Found', url=URL('https://api.vaillant-group.com/service-connected-control/vrc700/v1/emf/v2/<removed>/devices/<deviceID>/buckets?resolution=DAY&operationMode=COOLING&energyType=CONSUMED_ELECTRICAL_ENERGY&startDate=2024-02-08T00:00:00Z&endDate=2024-02-09T00:00:00Z')

**Comment by stephanbaron** [comment]
Same here with v0.7.2.
After re-initialization fetches the data once
Values on eBus get refreshed nevertheless.

ecoCOMPACT VSC 206/4-5 90
VR 921 (Firmware Version 0357.40.33)
VR 700 

Trace:

Logger: myPyllant.api
Source: /usr/local/lib/python3.12/site-packages/myPyllant/api.py:1027
First occurred: 16:38:54 (1 occurrences)
Last logged: 16:38:54

Could not get RTS data
Traceback (most recent call last):
  File "/usr/local/lib/python3.12/site-packages/myPyllant/api.py", line 1022, in get_rts
    response = await self.aiohttp_session.get(
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/aiohttp/client.py", line 691, in _request
    await raise_for_status(resp)
  File "/usr/local/lib/python3.12/site-packages/myPyllant/api.py", line 102, in on_raise_for_status
    response.raise_for_status()
  File "/usr/local/lib/python3.12/site-packages/aiohttp/client_reqrep.py", line 1060, in raise_for_status
    raise ClientResponseError(
aiohttp.client_exceptions.ClientResponseError: 404, message='Resource Not Found', url=URL('https://api.vaillant-group.com/service-connected-control/vrc700/v1/rts/52108911-9b9b-474e-bcad-6815b5d9f2a9/devices')

**Comment by signalkraft** [comment]
> I have an issue with hotwater boost it is possible to enable and disable it. But the switch goes automatically to off even oft the boost is still active.
> workaround: enable it again and immediately stop it.

Please try https://github.com/signalkraft/mypyllant-component/releases/tag/v0.7.3b2

**Comment by switschel** [comment]
> @stephanbaron you need the latest pre-release
> 
> @switschel do you get no energy entities at all? But they show up in the app? I'm not sure why that would happen.

No, no energy entities at all in HA. Yep, I can see them in the app.

Can I support you somehow?

**Comment by Rchtr** [comment]
Hello everyone,
I have also the problem that there is no shown entity for energy consumption.

`Error fetching myVAILLANT data: 404, message='Resource Not Found', url=URL('https://api.vaillant-group.com/service-connected-control/vrc700/v1/emf/v2/859917fb-83e3-491a-bc56-a901ca99139d/devices/92fbe10e-3858-5044-ae4c-ff15c93cfc7b/buckets?resolution=DAY&operationMode=DOMESTIC_HOT_WATER&energyType=CONSUMED_ELECTRICAL_ENERGY&startDate=2024-02-09T00:00:00Z&endDate=2024-02-10T00:00:00Z')`

Is there any solution?

**Comment by signalkraft** [comment]
> No, no energy entities at all in HA. Yep, I can see them in the app.
> 
> Can I support you somehow?


If you have an Android phone, are somewhat technically inclined, and have some time on your hands - then yes: https://signalkraft.com/mypyllant-component/3-reverse-engineering/

I would need to know what API endpoints your app is calling to get this data, because it's apparently doing something new ~~and exciting~~.

**Comment by switschel** [comment]
I discovered this URL and a few others. Let me know if you need additional ones. I clicked multipled times and recorded requests & responses.

```
GET /service-connected-control/end-user-app-api/v1/emf/v2/<removed>/devices/<removed>/buckets?resolution=DAY&operationMode=HEATING&energyType=CONSUMED_ELECTRICAL_ENERGY&startDate=2024-02-05T00%3A00%3A00%2B01%3A00&endDate=2024-02-12T00%3A00%3A00%2B01%3A00 HTTP/1.1
```

**Comment by signalkraft** [comment]
Let me know if https://github.com/signalkraft/mypyllant-component/releases/tag/v0.7.3b4 fixes things for you!

**Comment by thomasbendl57** [comment]
First of all - thanks for the integration and your effort on this!

My environment
VR920
VRC700
Version V0.7.3b4


My problem
After initialization and subsequent manual re-initialization fetches the data only once
But no further data updates afterwards

To me it looks like, it requests data from Vaillant site successfully - however I'm far away from being experienced

Sample log after retrieving data
--------------------------------
2024-02-11 14:09:50.052 DEBUG (MainThread) [custom_components.mypyllant] Finished fetching myVAILLANT data in 0.687 seconds (success: True)

Following error:
----------------
2024-02-11 14:09:50.082 ERROR (MainThread) [homeassistant] Error doing job: Task exception was never retrieved
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/update_coordinator.py", line 243, in _handle_refresh_interval
    await self._async_refresh(log_failures=True, scheduled=True)
  File "/usr/src/homeassistant/homeassistant/helpers/update_coordinator.py", line 399, in _async_refresh
    self.async_update_listeners()
  File "/usr/src/homeassistant/homeassistant/helpers/update_coordinator.py", line 182, in async_update_listeners
    update_callback()
  File "/usr/src/homeassistant/homeassistant/helpers/update_coordinator.py", line 479, in _handle_coordinator_update
    self.async_write_ha_state()
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 945, in async_write_ha_state
    self._async_write_ha_state()
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 1066, in _async_write_ha_state
    state, attr, capabilities, shadowed_attr = self.__async_calculate_state()
                                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 1003, in __async_calculate_state
    state = self._stringify_state(available)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 951, in _stringify_state
    if (state := self.state) is None:
                 ^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/components/climate/__init__.py", line 295, in state
    hvac_mode = self.hvac_mode
                ^^^^^^^^^^^^^^
  File "/config/custom_components/mypyllant/climate.py", line 427, in hvac_mode
    return [
           ^
IndexError: list index out of range

If you need further info pls shout...

**Comment by signalkraft** [comment]
Could you [generate test data](https://signalkraft.com/mypyllant-component/3-contributing/#contributing-test-data) please and attach it to this issue?

**Comment by Rchtr** [comment]
Version V0.7.3b4 has fixed it for me. Thank You.

**Comment by thomasbendl57** [comment]
> Could you [generate test data](https://signalkraft.com/mypyllant-component/3-contributing/#contributing-test-data) please and attach it to this issue?

Pls see attached

[Test_data_20240212_thomas.txt](https://github.com/signalkraft/mypyllant-component/files/14240897/Test_data_20240212_thomas.txt)

**Comment by GLehnhoff** [comment]
**All works fine**   
I just installed the Home Assistant myVAILLANT integration V0.7.3b5 as the Home Assistant Multimatic integration is not working anymore after I migrated on my mobile phone from the Multimatic app to the new myVAILLANT app. I installed the latest Beta version (as of the discussions ongoing here). I have a VR900 and an ecoCompact VSC 146 with multiMATIC 700f/4. All works fine.

**Comment by signalkraft** [comment]
@thomasbendl57 please check in https://github.com/signalkraft/mypyllant-component/issues/102 - this issue was originally about another problem that's been resolved, so I'm gonna close it.
